lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1881,7 +1885,14 @@ def cmd_set(*args)
         # [name, class] from payload_show_results
         value = mod.first
       end
+    end
 
+    unless valid_options.any? { |vo| vo.casecmp?(name) }
+      message = "Unknown datastore option: #{name}."
+      suggestion = DidYouMean::SpellChecker.new(dictionary: valid_options).correct(name).first
+      message << " Did you mean #{suggestion}?" if suggestion
+      print_error(message)
```

Comment:
Did we want to call `cmd_set_help` help here to keep this similar to the output above?

I'd also consider moving this all into its own function given the reuse of code here.